examples/systemcalls/systemcalls.c

```diff
@@ -93,7 +142,65 @@ bool do_exec_redirect(const char *outputfile, int count, ...)
  *
 */
 
-    va_end(args);
+    int out_fd = open(outputfile, O_WRONLY | O_TRUNC | O_CREAT, 0644);
+    if (out_fd == -1){
+    	printf("Error, issue opening output file!\r\n");
+    	va_end(args);
+    	return false;
+    }
+    	
+    int pid; 
+    int status; 
+    pid = fork();
+    
+    if (pid == -1){
+    	printf("Issue forking to create child process\r\n");
+    	va_end(args);
+    	return false;
+    }
+    
+    else if (pid == 0){
+    	//we are the child
+    	
+    	int dup_ret = 0; 
+    	
+    	dup_ret = dup2(out_fd, 1); //make file descriptor of our out_file mimic that of stdout (fd=1)
+    	if (dup_ret == -1){
+    		printf("Error redirecting stdout\r\n");
+    		va_end(args);
+    		return false;
+    	}
+    	
+    	
+    	execv(command[0],command);
+    	
+    	printf("Issue creating process with execv\r\n");
+    	exit (-1); //should not return from execv if successful
+    }
+    
+    if (waitpid (pid, &status, 0) == -1){
+    	printf("Error while waiting for child process\r\n");
+    	va_end(args);
+    	return false;
+    }
+    
+    
+    //we're parent if we made it here
+    close(out_fd);
+    if (waitpid (pid, &status, 0) == -1){
+    	printf("Error while waiting for child process\r\n");
+    	va_end(args);
+    	return false;
+    }
 
+    else if (WIFEXITED (status)){
+    	if (WEXITSTATUS(status) != 0){
+    		printf("Child process terminated with non-zero exit code!\r\n");
+    		va_end(args);
+    		return false;
+    	}
+    }	
+    
```

Comment:
See discussion in class about avoiding duplication between these two very similar functions.